shell/browser/api/electron_api_download_item.cc

```diff
@@ -71,65 +97,89 @@ DownloadItem::~DownloadItem() {
     download_item_->RemoveObserver(this);
     download_item_->Remove();
   }
+}
 
-  // Remove from the global map.
-  g_download_item_objects.erase(weak_map_id());
+bool DownloadItem::CheckDestroyed() const {
```

Comment:
_(nit)_

Slightly weird negative logic here where a method named "CheckDestroyed" returning false if the item is destroyed.

Not a blocker, but consider renaming to `CheckAlive` but keeping the logic the same? That way the usage would read like:

```cc
if (!CheckAlive())
  return;
if (download_item_->IsDone())) {
  ...
```

Reply:
This is matching a pattern I started in `electron_api_tray.h`, so I'll update both.